### PR TITLE
Bumped the packages 'Mesos' and 'mesos-modules'

### PR DESCRIPTION
**Pull request by sambatyon**
## High Level Description

This PR updates the `Mesos` and `mesos-modules` packages.

## Related Issues

* [CORE-1067](https://jira.mesosphere.com/browse/CORE-1067) Filter results of /master/slaves and the v1 call GET_AGENTS.
* [DCOS-17194](https://jira.mesosphere.com/browse/DCOS-17194) Rename `.cni` config files to `.conf`

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version(s) integrated: 
[dcos-mesos](https://github.com/mesosphere/mesos/compare/80928eaf1d66e2d06d4f301ac197c12d08eaa533...17979fea307b7c094c2bac8e5a3f25af33208a0c)
[dcos-mesos-modules](https://github.com/dcos/dcos-mesos-modules/compare/2a5a02f1de47dcbe44844f7358adf85ea630cf4a...4b2432429e5a283a4bdf26a023d2327089315eb4)
  - [x] Test Results: [link to CI job test results for component]
[Mesos CI](https://jenkins.mesosphere.com/service/jenkins//job/mesos/job/Mesos_CI-build/1468)
[dcos-mesos-modules](https://jenkins.mesosphere.com/service/jenkins/job/mesos/job/modules/job/DCOS_Modules_CI-nightly/203/)
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

**Comment by sambatyon** [comment]
After debugging the broken test `TestUpgrade.test_mesos_task_state_remains_consistent`, my best informed guess is that it is a broken test. I cannot confirm a 100% because the tests gives very little debug info and at this time `dcos-test-utils` is going through some kind of refactoring and the repository used by DC/OS at the moment is read only.

To reach my conclusion what I did then was the following:
1. Create local copies of the versions of Mesos used in 1.9.1 (the one used in upgrade) and the one used in this PR.
1. Launched a cluster, one agent, one master of `mesos-dcos-1.91.`.
1. Launched a task in the cluster.
1. Get the `mesos/state` of the cluster at this point in time.
1. Restarted the master and launched one of `mesos-dcos-pull-1753`. Do the same with the agent.
1. Wait until task reconciliation finishes.
1. Get the `mesos/state` of the cluster at this point in time.
1. Compare the JSON of the tasks.

For the last part I used the following script:

```py
task_state_start = get_task_state('mesos-dcos-1.9.1-state.json')
task_state_end = get_task_state('mesos-dcos-master-state.json')

print([item in task_state_end.items() for item in task_state_start.items()])

for item in task_state_start.items():
    if item not in task_state_end.items():
        print('{} is not in task_state_end'.format(item))
```

which performs the same comparison as `TestUpgrade` but it will print which one is the offending entry. The different one is the `statuses` entry. The main problem here is that the comparison used not only compares the keys but also the values. Now `statuses.network_infos.ip_addresses` includes also a field called `protocol` which is inexistent in previous version of DC/OS and it is enough to make the comparison end.

My proposed solution to fix the test is to modify it so it doesn't do a comparison of the keys, but that it does a recursive is contained operation. This test attempts that but stops short of the first layer.

@mellenburg 
@jgehrcke

File: packages/mesos-modules/buildinfo.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "single_source" : {
       "kind": "git",
       "git": "https://github.com/dcos/dcos-mesos-modules.git",
-      "ref": "2a5a02f1de47dcbe44844f7358adf85ea630cf4a",
+      "ref": "1708f8acf39d0cadfb8a4735f607296c43a87a8c",
```
Comment (adam-mesos) [inline packages/mesos-modules/buildinfo.json]: Please use latest master to also include the fix for https://jira.mesosphere.com/browse/CORE-1137

**Comment by sambatyon** [comment]
Finally got the test suite to pass. It does requires that [advanced-tests:pull/2](https://github.com/mesosphere/advanced-tests/pull/2) is merged beforehand.

CC @mellenburg

**Comment by adam-mesos** [comment]
@sambatyon Can you please update this PR and get it across the finish line? DC/OS 1.10.0-beta1 is shipping with a pre-1.3.0 Mesos (225 commits behind master!), and I promised beta2 would be better.
I see #1755 and #1783 trying to bump Mesos+modules, but they don't have your magic touch.
Up to you if you want to just rebase with the same Mesos SHA, or try something closer to master.
cc: @vinodkone

**Comment by jgehrcke** [comment]
`integration-test/dcos-docker` as well as `smoke-test/dcos-docker` failed as of https://jira.mesosphere.com/browse/DCOS_OSS-1481 -- known issue.

**Comment by jgehrcke** [comment]
@vinodkone and @adam-mesos now that the CI looks reasonably well -- can we get two reviews here?

**Comment by sambatyon** [comment]
@asridharan you seem to own the dcos-mesos-modules repo, can you give a view at this dcos mesos bump.
@greggomann could you also give it a quick view?

**Comment by asridharan** [comment]
@sambatyon sorry didn't follow which Mesos dump are you referring to?

**Comment by jgehrcke** [comment]
> @sambatyon sorry didn't follow which Mesos dump are you referring to?

@asridharan Alexander is asking if you can review this PR which includes this Mesos bump:
```
-    "ref": "80928eaf1d66e2d06d4f301ac197c12d08eaa533",
 -    "ref_origin" : "dcos-mesos-master-013f7e21"
 +    "ref": "17979fea307b7c094c2bac8e5a3f25af33208a0c",
 +    "ref_origin" : "dcos-mesos-master-1b15ff7a"
```

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

File: packages/mesos/README.md [hunk 2]
```diff
@@ -2,8 +2,7 @@
...
 
-<li>[103c78c43acd7df0fe7252963333be00e7aef02b] Set LIBPROCESS_IP into docker container.
-<li>[202dc73803a3e681e31dd30ca68a44fdae2ba902] Mesos UI: Change paths, pailer, and logs to work with a reverse proxy.
-<li>[6bbab709eb09aeecac705ff4ce5e5867bbe7df30] Revert "Fixed the broken metrics information of master in WebUI."
-<li>[a9678f06a8e222d97b1f784edf96030beb643552] Updated mesos containerizer to ignore GPU isolator creation failure.
-<li>[80928eaf1d66e2d06d4f301ac197c12d08eaa533] Changed an absolute path to relative in the Mesos UI.
+<li>[20e5654a294f599b637efadf32206e7b15398081] Set LIBPROCESS_IP into docker container.
+<li>[098743933a99925cfccb19f132393fa9f236c31e] Mesos UI: Change paths, pailer, and logs to work with a reverse proxy.
+<li>[f01e1ecc7ee9e9879d7059f8862a389f9f9c25c6] Revert "Fixed the broken metrics information of master in WebUI."
+<li>[17979fea307b7c094c2bac8e5a3f25af33208a0c] Updated mesos containerizer to ignore GPU isolator creation failure.
```
Comment (adam-mesos) [inline packages/mesos/README.md]: This version of the "Updated mesos containerizer to ignore GPU isolator creation failure." cherry-pick now includes the minor "Changed an absolute path to relative in the Mesos UI." that we used to include separately.

**Pull request merged**